Use `mkstemp` instead of `tmpnam` in `sstregistertool.cc`

As part of #983 I unwittingly used a deprecated function:
```
/Users/ejberqu/development/sst/github/sst-core/src/sst/core/sstregistertool.cc:153:39: warning: 'tmpnam' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of tmpnam(3), it is highly recommended that you use mkstemp(3) instead. [-Wdeprecated-declarations]
    const auto        tempfile = std::tmpnam(nullptr);
                                      ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdio.h:189:1: note: 'tmpnam' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of tmpnam(3), it is highly recommended that you use mkstemp(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
```

--- a/src/sst/core/sstregistertool.cc
+++ b/src/sst/core/sstregistertool.cc
@@ -149,8 +149,14 @@ void
 sstUnregister(const std::string& element)
 {
     // setup element names to look for
-    const std::string str1     = START_DELIMITER + element + STOP_DELIMITER;
-    const auto        tempfile = std::tmpnam(nullptr);
+    const std::string str1             = START_DELIMITER + element + STOP_DELIMITER;
+    char              tempfilename[24] = "sstsimulator.confXXXXXX";
+    const auto        fd               = mkstemp(&tempfilename[0]);
+    if ( fd == -1 ) {
+        std::cerr << "\tError creating temporary file for writing config with unregistered element " << element << "\n";
+        return;
+    }
+    const std::string tempfile(tempfilename);
 
     std::string s     = "";
     int         found = 0;
@@ -184,7 +190,7 @@ sstUnregister(const std::string& element)
     infile  = std::ifstream(tempfile, std::ios::binary);
     outfile = std::ofstream(cfgPath, std::ios::binary);
     outfile << infile.rdbuf();
-    if ( std::remove(tempfile) != 0 ) {
+    if ( std::remove(tempfile.c_str()) != 0 ) {
         std::cerr << "\tError removing " << tempfile << " after moving updated config\n";
         return;
     }

